activerecord/lib/active_record/relation.rb

```diff
@@ -708,5 +706,19 @@ def tables_in_string(string)
         # ignore raw_sql_ that is used by Oracle adapter as alias for limit/offset subqueries
         string.scan(/([a-zA-Z_][.\w]+).?\./).flatten.map(&:downcase).uniq - ["raw_sql_"]
       end
+
+      def values_for_create(attributes = nil)
+        result = attributes ? where_values_hash.merge!(attributes) : where_values_hash
+
+        # NOTE: if there are same keys in both create_with and result, create_with should be used.
+        # This is to make sure nested attributes don't get passed to the klass.new,
+        # while keeping the precedence of the duplicate keys in create_with.
+        # Sorry, this doesn't look nice but it's the only way now. :(
```

Comment:
Can you remove this line?
This doesn't look nice, but it will be resolved in the near future by refactoring internal scoping things.

Reply:
just line 716 or the entire comment?